tests/minifier.js

```diff
@@ -3353,6 +3353,11 @@ QUnit.test('decode entity characters', function(assert) {
   assert.equal(minify(input, { decodeEntities: false }), input);
   assert.equal(minify(input, { decodeEntities: true }), input);
 
+  // https://github.com/kangax/html-minifier/issues/964
+  input = '&amp;xxx; &amp;xxx &ampthorn; &ampthorn &ampcurren;t &ampcurrent';
+  output = '&ampxxx; &xxx &ampthorn; &ampthorn &ampcurren;t &ampcurrent';
```

Comment:
Without this patch, the output is:

```js
'&ampxxx; &xxx &ampampthorn; &thorn &ampampcurren;t &current'
```

…which renders as:

```
&xxx; &xxx &ampthorn; þ &ampcurren;t ¤t
```

…whereas the input HTML renders as:

```
&xxx; &xxx &thorn; &thorn &curren;t &current
```